src/app.js

```diff
@@ -5,40 +5,23 @@ import 'normalize.css/normalize.css';
 
 import AppRouter from './router/AppRouter';
 import configureStore from './store/configureStore';
-import { addActivity } from './store/activities/actions';
-import { MONDAY, TUESDAY } from './constants/dates';
+import { startSetActivities } from './store/activities/actions';
 import './styles/styles.scss';
 import './firebase/firebase';
 
 const store = configureStore();
 
-// TODO: dummy data to be removed
-store.dispatch(
-  addActivity({
-    name: 'matma',
-    day: TUESDAY,
-    classNo: '3',
-    room: 'gym',
-    teacher: 'Indiana Jones',
-  }),
-);
-store.dispatch(
-  addActivity({
-    name: 'polski',
-    day: MONDAY,
-    classNo: '1',
-    room: 'hell',
-    teacher: 'Malski',
-  }),
-);
-
 const jsx = (
   <Provider store={store}>
     <AppRouter />
   </Provider>
 );
 
-ReactDOM.render(jsx, document.getElementById('app-root'));
+ReactDOM.render(<p>Loading...</p>, document.getElementById('app-root'));
+
+store.dispatch(startSetActivities()).then(() => {
```

Comment:
I cannot imagine that it will stay in this form. 

For sure, first fetching will be done when user will sign in, but still... You shouldn't manage some kind of loading in app initialization. So - even if temporary, there should be component which handles fetching initial data. In this file, there shouldnt be `store.dispatch` at all